qiskit/visualization/dag_visualization.py

```diff
@@ -166,7 +224,14 @@ def edge_attr_func(edge):
     dot = pydot.graph_from_dot_data(dot_str)[0]
 
     if filename:
+        if "." not in filename:
+            raise InvalidFileError("Parameter 'filename' must be in format 'name.extension'")
         extension = filename.split(".")[-1]
+        if extension not in FILENAME_EXTENSIONS:
```

Comment:
`FILENAME_EXTENSIONS` is a list, so `not in` will be a little slow, considering it's got quite a few elements.  It'll be better if we make that a set, then this line will be fast.

Reply:
Wow! You are right. Thank you.